source/TWA/Thesis/Chapter6/SequenceContinuity.lagda.md

```diff
@@ -0,0 +1,200 @@
+open import UF.Miscelanea
+open import UF.Equiv
+
+module TWA.Thesis.Chapter6.SequenceContinuity (fe : FunExt) where
+
+open import TWA.Thesis.Chapter2.Sequences
+open import TWA.Thesis.Chapter3.ClosenessSpaces fe
+open import TWA.Thesis.Chapter3.ClosenessSpaces-Examples fe
+
+open import MLTT.Two-Properties
+
+seq-f-ucontinuous¹ : {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+                   → (f : (ℕ → X) → (ℕ → Y))
+                   → 𝓤 ⊔ 𝓥 ̇
+seq-f-ucontinuous¹ {𝓤} {𝓥} {X} f
+ = (ϵ : ℕ) → Σ δ ꞉ ℕ , ((x₁ x₂ : (ℕ → X))
+ → (x₁ ∼ⁿ x₂) δ → (f x₁ ∼ⁿ f x₂) ϵ)
+
+seq-f-ucontinuous² : {X : 𝓤 ̇ } {Y : 𝓥 ̇ } {Z : 𝓦 ̇ }
+                   → (f : (ℕ → X) → (ℕ → Y) → (ℕ → Z))
+                   → 𝓤 ⊔ 𝓥 ⊔ 𝓦  ̇
+seq-f-ucontinuous² {𝓤} {𝓥} {𝓦} {X} {Y} f
+ = (ϵ : ℕ) → Σ (δˣ , δʸ) ꞉ ℕ × ℕ ,
+   ((x₁ x₂ : (ℕ → X)) (y₁ y₂ : (ℕ → Y))
+ → (x₁ ∼ⁿ x₂) δˣ → (y₁ ∼ⁿ y₂) δʸ → (f x₁ y₁ ∼ⁿ f x₂ y₂) ϵ)
+
+map-ucontinuous' : {X : 𝓤 ̇ } {Y : 𝓥 ̇ } 
+                 → (f : X → Y) → seq-f-ucontinuous¹ (map f)
+map-ucontinuous' f ε = ε , λ α β α∼ⁿβ k k<ε → ap f (α∼ⁿβ k k<ε)
+
+zipWith-ucontinuous' : {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+                     → (f : X → X → Y)
+                     → seq-f-ucontinuous² (zipWith f)
+zipWith-ucontinuous' f ε
+ = (ε , ε)
+ , (λ α₁ α₂ β₁ β₂ α∼ β∼ k k<ϵ
+    → ap (λ - → f - (β₁ k)) (α∼ k k<ϵ)
+    ∙ ap (f (α₂ k)) (β∼ k k<ϵ))
+
+seq-f-ucontinuous²-left : {X : 𝓤 ̇ } {Y : 𝓥 ̇ } {Z : 𝓦 ̇ }
+                        → (f : (ℕ → X) → (ℕ → Y) → (ℕ → Z))
+                        → seq-f-ucontinuous² f
+                        → (β : ℕ → Y)
+                        → seq-f-ucontinuous¹ (λ α → f α β)
+seq-f-ucontinuous²-left f ϕ β ε
+ = pr₁ (pr₁ (ϕ ε))
+ , λ α₁ α₂ α∼ → pr₂ (ϕ ε) α₁ α₂ β β α∼ (λ _ _ → refl)
+
+seq-f-ucontinuous²-right : {X : 𝓤 ̇ } {Y : 𝓥 ̇ } {Z : 𝓦 ̇ }
+                         → (f : (ℕ → X) → (ℕ → Y) → (ℕ → Z))
+                         → seq-f-ucontinuous² f
+                         → (α : ℕ → X)
+                         → seq-f-ucontinuous¹ (f α)
+seq-f-ucontinuous²-right f ϕ α ε
+ = pr₂ (pr₁ (ϕ ε))
+ , λ β₁ β₂ → pr₂ (ϕ ε) α α β₁ β₂ (λ _ _ → refl)
+
+seq-f-ucontinuous²-both : {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+                        → (f : (ℕ → X) → (ℕ → X) → (ℕ → Y))
+                        → seq-f-ucontinuous² f
+                        → seq-f-ucontinuous¹ (λ α → f α α)
+seq-f-ucontinuous²-both f ϕ ε
+ = δ
+ , λ α β α∼ᵐβ → pr₂ (ϕ ε) α β α β
+     (λ i i<m → α∼ᵐβ i
+       (<-≤-trans i δ₁ δ i<m (max-≤-upper-bound  δ₁ δ₂)))
+     (λ i i<m → α∼ᵐβ i
+       (<-≤-trans i δ₂ δ i<m (max-≤-upper-bound' δ₂ δ₁)))
+ where
+  δ₁ δ₂ δ : ℕ
+  δ₁ = pr₁ (pr₁ (ϕ ε))
+  δ₂ = pr₂ (pr₁ (ϕ ε))
+  δ  = max δ₁ δ₂
+
+seq-f-ucontinuous²-comp
+ : {X : 𝓤 ̇ } {Y : 𝓥 ̇ } {Z : 𝓦 ̇ } {W : 𝓣 ̇ } {T : 𝓤' ̇ }
+ → (f : (ℕ → X) → (ℕ → W) → (ℕ → T))
+ → (g : (ℕ → Y) → (ℕ → Z) → (ℕ → W))
+ → seq-f-ucontinuous² f
+ → seq-f-ucontinuous² g
+ → (z : ℕ → Z) → seq-f-ucontinuous² λ x y → f x (g y z)
+seq-f-ucontinuous²-comp
+ {_} {_} {_} {_} {_} {X} {Y} {Z} {W} {T} f g ϕᶠ ϕᵍ z ϵ = δ , γ
```

Comment:
Instead of all the `{_}`, you can just name the implicit arguments like `{X = X}`, and even  do `f {X = X} {Y} {Z} ...`. I don‘t think it’s good style to have all these `{_}`.

Reply:
No, please don't use {X = X}.

(1) It is inelegant and doesn't make sense to people who know type theory but not Agda.
(2) It is not robust. If people change the original definition to rename the variables then the calls don't work any more.

In the next version of Agda there is a solution to this problem (I am not sure it is very good either).